test/functional/Java22andUp/src/org/openj9/test/jep454/upcall/UpcallMethodHandles.java

```diff
@@ -542,8 +915,7 @@ public static MemorySegment createNewCharFromCharAndCharFromPtr_RetPtr(MemorySeg
 		int diff = (charArg2 >= charArg1) ? (charArg2 - charArg1) : (charArg1 - charArg2);
 		diff = (diff > 5) ? 5 : diff;
 		char result = (char)(diff + 'A');
-		MemorySegment resultSegmt = arena.allocate(JAVA_CHAR);
-		resultSegmt.set(JAVA_CHAR, 0, result);
+		MemorySegment resultSegmt = arena.allocateFrom(JAVA_CHAR, (char)result);
```

Comment:
Please remove the redundant cast to `char`. Lines 948, 976 and 1075 also have redundant casts.